File: flink/src/main/java/com/zoo/flink/java/source/SocketSourceDemo.java
package com.zoo.flink.java.source;

import com.zoo.flink.java.util.FlinkEnv;
import org.apache.flink.streaming.api.datastream.DataStream;

/**
 * @Author: JMD
 * @Date: 5/10/2023
 */
public class SocketSourceDemo extends FlinkEnv {
    public static void main(String[] args) throws Exception {
        // 从socket中读取数据生成流
        DataStream<String> stream = env.socketTextStream("hadoop102", 7777);

        stream.print();
        env.execute();
    }
}

File: flink/src/main/scala/com/zoo/flink/scala/source/SocketSourceDemo.scala
package com.zoo.flink.scala.source

import com.zoo.flink.scala.FlinkEnv
import org.apache.flink.streaming.api.scala.createTypeInformation

/**
 * Author: JMD
 * Date: 5/11/2023
 */
object SocketSourceDemo extends FlinkEnv{
  def main(args: Array[String]): Unit = {
    val inputDataStream = env.socketTextStream("localhost", 7777)

    //对数据进行转换处理
    val resultDataStream = inputDataStream.flatMap(_.split(" ")).filter(_.nonEmpty).map((_, 1))
      .keyBy(_._1)
      .sum(1)

    //打印输出//打印输出
    resultDataStream.print
  }
}
